packages/protocol/contracts/AccountRegistry/epochs/20200220/Behaviour20200220.sol

```diff
@@ -0,0 +1,146 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+import "../20200207/Behaviour20200207.sol";
+import "../../../interfaces/IERC20Permit.sol";
+
+/**
+ * @title Behaviour20200220 implementation
+ * @author AZTEC
+ * @dev This behaviour contract overloads the deposit() account registry method, with a deposit() 
+ * implementation that is compatible with the DAI permit() function.
+ * 
+ * Note the behaviour contract version naming convention is based on the date on which the contract
+ * was created, in the format: YYYYMMDD
+ * 
+ * Copyright 2020 Spilsbury Holdings Ltd 
+ *
+ * Licensed under the GNU Lesser General Public Licence, Version 3.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ **/
+contract Behaviour20200220 is Behaviour20200207 {
+    /**
+    * @dev epoch number, used for version control in upgradeability. The naming convention is based on the
+    * date on which the contract was created, in the format: YYYYMMDD
+    */
+    uint256 public epoch = 20200220;
+
+    /**
+    * @dev This deposit() implementation performs the standard AccountRegistry deposit functionality, but it
+    * is also specifically compatible with the DAI permit() function. 
+    
+    * The permit() function removes the need for a user to send a seperate approval() transaction
+    * beforehand in order to approve another address to spend DAI on their behalf. 
+    * 
+    * @param _registryOwner - owner of the zkAsset
+    * @param _owner - owner of the ERC20s being deposited
+    * @param _proofHash - hash of the zero-knowledge deposit proof
+    * @param _proofData - cryptographic data associated with the zero-knowledge proof
+    * @param _value - number of ERC20s being deposited
+    * @param signature - EIP712 signature supplied to permit() for granting approval
+    * @param nonce - permit nonce used for signature replay protection
+     */
+    function deposit(
```

Comment:
As discussed, should make the permit call to transfer tokens to this contract first, then rely on `super.deposit()` to call existing deposit function in previous inherited behaviour to make changes explicit, and increase confidence that no changes in logic have occurred beyond expected ones.

Reply:
Yeah good idea, have done